### ISSUE TITLE
Add a function to get plain object from $data

### ISSUE THREAD

**Issue by michaelsogos** [vuejs/vue#3833]
=========== FEATURE REQUEST ==============
======== GET PLAIN JSON FROM $data ==========

Because we use an application with Bootstrap, and because Bootstrap is dependant on JQuery, we don't see the reason why to add a new external library like vue-resources to just having an AJAX helper.
Basically we comes from Ractive.JS and some other framework that basically have kind of function to get POJO from view-model object in a way that is easy to use in any situation, specially in $.ajax.

1. Because we guess that jQuery library is used very often
2. Because a readable method to get plain object from $data is better than JSON.parse(...)
3. Because from code point of view the snippet JSON.parse(JSON.stringify(...)) sounds like a workaround
4. Because we study for a better performant deep cloner of $data

And for many other reason that at the moment we don't figure out :), we suggest to implement a function that simply return a not reactive copy of $data.

```javascript
function deepClone(obj){
  var result = {};
  var props = Object.keys(obj);
  props.forEach(function(propName) {  
      var clonedValue = obj[propName];
      if(typeof clonedValue === 'object'){
            clonedValue = deepClone(obj[propName]); 
      }        
      result[propName] = clonedValue;
  });
  return result;
}
```

Click this [Test](https://jsfiddle.net/michaelsogos/w6m1x5ve/) to get better performance result than JSON.parse(JSON.stringify(...))

### Vue.js version
2.0.1

### Reproduction Link
https://jsfiddle.net/michaelsogos/w6m1x5ve/

**Comment by fnlctrl** [comment]
Hi, thanks for filling this issue.

>  to get POJO from view-model object in a way that is easy to use in any situation, specially in $.ajax

You can just pass the reactive data (obj with getters/setters) to any library (for example $.ajax) and they will work as if the data were POJO. I guess there could hardly be any practical advantage in cloning before passing.

If you do need a deep clone implementation with decent performance I'd suggest lodash/underscore, but that's probably not vue's concern.

(Side note: If you really care about performance, you should probably avoid using Object.keys and Array prototype functions to just loop over an object... https://jsfiddle.net/0poxoe90/)

**Comment by yyx990803** [comment]
Why do you want to clone it for $.ajax? I don't see a reason. The converted object works exactly the same for serialization, I don't see the point of cloning.

**Comment by michaelsogos** [comment]
Ok, probably i need to test it more, but if i give vue.$data directly to jQuery ajax, it will throw an exception.
So i look for a solution in internet and the most found answer is to use JSON.parse(...), also it is mentioned on vue api web site.
That's why i try to found a better solution to get plain object, because i guess that the cause of the jQuery exception are getter and setter.
I will try next days to create an jsfiddler example with vue and jquery to prove my issue.

For the project we are working on, lodash and underscore are not needed ;), but i keep in mind this suggestion.

The above js code is just a suggestion of improved code than JSON.parse(...), object.keys work faster than for(prop in obj) with an if(obj[prop].hasOwnProperty), but we can make more test.

**Comment by michaelsogos** [comment]
Ok, I tested with my team what @fnlctrl suggest, and it better in performance, so thanks for deepClone2 :)

**Comment by michaelsogos** [comment]
Ok @yyx990803, we tested jQuery 3.1.1 ajax helper giving $data object, and it works perfectly.
I really not sure what happends some days ago, but for some reason jQuery give back an exception, now seems to not appear anymore.
